### ISSUE TITLE
no-mixed-operators: don't require parentheses around a single ternary operator argument

### ISSUE THREAD

**Issue by akwodkiewicz** [eslint/eslint#14223]
The only way to write the `a ? b && c : d` expression in compliance with `no-mixed-operators` rule (when ternary operator is added to logical operators group) is to use the parentheses **twice**:

```js
(a) ? (b && c) : d
```

In my opinion, parentheses around `a` shouldn't be required.

**What rule do you want to change?**
no-mixed-operators

**Does this change cause the rule to produce more or fewer warnings?**
Fewer warnings

**How will the change be implemented? (New option, new default behaviour, etc.)?**
New default behaviour, I guess

**Please provide some example code that this change will affect:**

Group: `["&&", "||", "?:"]`

```js
a ? (b && c) : d
```

**What does the rule currently do for this code?**
Reports an error

**What will the rule do after it's changed?**
It will not report an error

**Are you willing to submit a pull request to implement this change?**
No

**Comment by mdjermanovic** [comment]
Hi @akwodkiewicz, thanks for the issue!

This is a bug, the rule requires additional parentheses around `test` or the whole `ConditionalExpression` for no reason.

```js
/* eslint no-mixed-operators: ["error", { groups: [["&&", "||", "?:"]] }] */

a ? b && c : d; // error

a ? (b && c) : d; // error, false positive

(a) ? (b && c) : d; // no error

(a ? (b && c) : d); // no error
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLW1peGVkLW9wZXJhdG9yczogW1wiZXJyb3JcIiwgeyBncm91cHM6IFtbXCImJlwiLCBcInx8XCIsIFwiPzpcIl1dIH1dICovXG5cbmEgPyBiICYmIGMgOiBkOyAvLyBlcnJvclxuXG5hID8gKGIgJiYgYykgOiBkOyAvLyBlcnJvciwgZmFsc2UgcG9zaXRpdmVcblxuKGEpID8gKGIgJiYgYykgOiBkOyAvLyBubyBlcnJvclxuXG4oYSA/IChiICYmIGMpIDogZCk7IC8vIG5vIGVycm9yXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

Marked as accepted, I'll work on this.
